language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/MappingContextProvider.java

```diff
@@ -187,7 +194,95 @@ protected boolean hasReadonlyKW(Node evalNodeAtCursor) {
+        List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition()).stream()
+                .filter(symbolFilter.and(symbol -> {
+                    Optional<TypeSymbol> typeDescriptor = SymbolUtil.getTypeDescriptor(symbol);
+                    return typeDescriptor.isPresent() && typeDescriptor.get().subtypeOf(resolvedType.get())
+                            && (CommonUtil.getRawType(typeDescriptor.get()).typeKind() == TypeDescKind.MAP
+                            || CommonUtil.getRawType(typeDescriptor.get()).typeKind() != TypeDescKind.RECORD);
+                })).collect(Collectors.toList());
+
+        return getSpreadFieldCompletionItemList(visibleSymbols, context);
+    }
+
+    private List<LSCompletionItem> getSpreadFieldCompletionItems(BallerinaCompletionContext context,
+                                                                 List<RecordFieldSymbol> validFields) {
+        Predicate<Symbol> symbolFilter = this.getVariableFilter().or(symbol -> (symbol.kind() == FUNCTION));
+        List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition()).stream()
+                .filter(symbolFilter.and(symbol -> {
+                    Optional<TypeSymbol> typeDescriptor = SymbolUtil.getTypeDescriptor(symbol);
+                    if (typeDescriptor.isEmpty()) {
+                        return false;
+                    }
+                    TypeSymbol rawType;
+                    if (typeDescriptor.get().typeKind() == TypeDescKind.FUNCTION) {
+                        Optional<TypeSymbol> returnTypeSymbol =
+                                ((FunctionTypeSymbol) typeDescriptor.get()).returnTypeDescriptor();
+                        if (returnTypeSymbol.isEmpty()) {
+                            return false;
+                        }
+                        rawType = CommonUtil.getRawType(returnTypeSymbol.get());
+                    } else {
+                        rawType = CommonUtil.getRawType(typeDescriptor.get());
+                    }
+                    if (rawType.typeKind() != TypeDescKind.RECORD) {
+                        return false;
+                    }
+                    Map<String, RecordFieldSymbol> recordFields = ((RecordTypeSymbol) rawType).fieldDescriptors();
+                    for (RecordFieldSymbol fieldSymbol : recordFields.values()) {
+                        if (fieldSymbol.getName().isEmpty() || validFields.stream()
+                                .filter(validField -> validField.getName().isPresent()
+                                        && validField.getName().get().equals(fieldSymbol.getName().get())
+                                        && fieldSymbol.typeDescriptor()
+                                        .subtypeOf(validField.typeDescriptor())).findAny().isEmpty()) {
+                            return false;
+                        }
+                    }
+                    return true;
```

Comment:
We have multiple returns which is okay, but given the logic it is bit complex to understand. Check whether if we can simplify this